File: examples/simple_robot/python_output/robot.py
import random

state = {'x' : random.randint(0, 4),
         'y' : random.randint(0, 4),
         'target_x' :  random.randint(0, 4),
         'target_y' :  random.randint(0, 4),
         'missions_remaining' : random.randint(1, 10)}


def get_mission():
    if random.choice([True, False]):
        return (True, state['target_x'], state['target_y'])
    else:
        return (False, random.randint(0, 4), random.randint(0, 4))


def get_position():
    return (state['x'], state['y'])


def go_right():
    state['x'] = state['x'] + 1
    check_completion()
    return


def go_left():
    state['x'] = state['x'] - 1
    check_completion()
    return


def go_up():
    state['y'] = state['y'] + 1
    check_completion()
    return


def go_down():
    state['y'] = state['y'] - 1
    check_completion()
    return


def check_completion():
    if state['missions_remaining'] >= 0:
        if state['x'] == state['target_x'] and state['y'] == state['target_y']:
            state['missions_remaining'] = state['missions_remaining'] - 1
            state['target_x'] = random.randint(0, 4)
            state['target_y'] = random.randint(0, 4)
    return

File: examples/simple_robot/haskell/simple_robot/app/Robot.hs
module Robot where
import BehaviorTreeCore
import BTreeHaveMission
import BTreeTargetReached
import BTreeXTooSmall
import BTreeXTooBig
import BTreeYTooSmall
import BTreeYTooBig
import BTreeGetMission
import BTreeGetPosition
import BTreeClearMission
import BTreeGoRight
import BTreeGoLeft
import BTreeGoUp
import BTreeGoDown
bTreeNodeGetPosition = BTreeNode getPosition [] 1
bTreeNodeTargetReached = BTreeNode targetReached [] 4
bTreeNodeClearMission = BTreeNode clearMission [] 5
bTreeNodeResetCompletedMission = BTreeNode sequenceFunc [bTreeNodeTargetReached, bTreeNodeClearMission] 3
bTreeNodeResetCompletedMissionFis = BTreeNode (decoratorCreator (xISyCreator Failure Success)) [bTreeNodeResetCompletedMission] 2
bTreeNodeHaveMission = BTreeNode haveMission [] 7
bTreeNodeGetMission = BTreeNode getMission [] 8
bTreeNodeConfirmMission = BTreeNode selectorFunc [bTreeNodeHaveMission, bTreeNodeGetMission] 6
bTreeNodeXTooSmall = BTreeNode xTooSmall [] 11
bTreeNodeGoRight = BTreeNode goRight [] 12
bTreeNodeTryRight = BTreeNode sequenceFunc [bTreeNodeXTooSmall, bTreeNodeGoRight] 10
bTreeNodeXTooBig = BTreeNode xTooBig [] 14
bTreeNodeGoLeft = BTreeNode goLeft [] 15
bTreeNodeTryLeft = BTreeNode sequenceFunc [bTreeNodeXTooBig, bTreeNodeGoLeft] 13
bTreeNodeYTooSmall = BTreeNode yTooSmall [] 17
bTreeNodeGoUp = BTreeNode goUp [] 18
bTreeNodeTryUp = BTreeNode sequenceFunc [bTreeNodeYTooSmall, bTreeNodeGoUp] 16
bTreeNodeYTooBig = BTreeNode yTooBig [] 20
bTreeNodeGoDown = BTreeNode goDown [] 21
bTreeNodeTryDown = BTreeNode sequenceFunc [bTreeNodeYTooBig, bTreeNodeGoDown] 19
bTreeNodeMoveRobot = BTreeNode selectorFunc [bTreeNodeTryRight, bTreeNodeTryLeft, bTreeNodeTryUp, bTreeNodeTryDown] 9
bTreeNodeRobotControl = BTreeNode sequenceFunc [bTreeNodeGetPosition, bTreeNodeResetCompletedMissionFis, bTreeNodeConfirmMission, bTreeNodeMoveRobot] 0
